apps/chat_server/lib/chat_server/contexts/starred_messages/get_starred_messages.ex

```diff
@@ -0,0 +1,11 @@
+defmodule ChatServer.GetStarredMessages do
+  alias ChatServer.Schema
+
+  def call(user) do
+    %{starred_messages: get_starred_messages(user)}
+  end
+
+  defp get_starred_messages(user) do
+    Schema.StarredMessage.get_by_user(user)
+  end
+end
```

Comment:
This context looks good. Two thoughts to take it further:

1. I've been refactoring context objects to always return a tuple, e.g. either `{:ok, result}` or `{:error, msg}`. This makes it a lot easier to chain context objects in `case` and `with` statements and ensure a valid result is being returned without introducing lots of `nil` checks downstream. For an example of that, see: [GetSubscription](https://github.com/oberon-chat/chat-server/blob/master/apps/chat_server/lib/chat_server/contexts/subscriptions/get_subscription.ex).

2. A context object could be called from multiple places, e.g. Websocket and GraphQL. The returned `%{starred_messages: data}` map is a Websocket implementation detail, and should be handled in the channel file instead of in the context. The context should return just the data, or ideally, `{:ok, data}` | `{:error, msg}`